### ISSUE TITLE
High CPU Usage w/ Sumneko Lua

### ISSUE THREAD

**Issue by jonleopard** [ChristianChiarulli/Neovim-from-scratch#98]
Been following along your great series over at youtube. I noticed when editing .lua files, the neovim gets very sluggish and my CPU fans start to whirl up.  I checked the LSP logs and this showed up:

```
 [WARN][2022-03-04 11:00:21] .../lua/vim/lsp.lua:751 "server_request: no handler found for"  "workspace/semanticTokens/refresh"
```
Here's a look at the process, it stays at 99.5% as long as the .lua buffer is open:
<img width="716" alt="Screenshot 2022-03-04 at 15 07 53" src="https://user-images.githubusercontent.com/2111234/156848463-56ef5f32-2e52-432d-94a0-879bcaa0f785.png">

Any ideas?

**Comment by sazro89** [comment]
Unsure of if this will help you out, but I was suffering from workspaces never being able to fully load, and lua-language-server using 100%+ cpu as well as multiple gigabytes of ram as i left it running for longer.  I delved a bit into the problem, and  [this post](https://www.reddit.com/r/neovim/comments/pi1nw3/comment/hbmtah8/?utm_source=share&utm_medium=web2x&context=3) led me to looking through the default server configuration located at `~/.local/share/nvim/site/pack/packer/start/nvim-lspconfig/lua/lspconfig/server_configurations/sumneko_lua.lua`.  

The tl;dr is that sumneko will recursively go to parent directories until it sees certain files which it uses to determine the scope of a project.  If you're following along by hand and manually copying files (like I was) there's no .git folder that sumneko will find and as a result it will keep trying to pull in more and more code to parse.  So I made a new git project in `~/.config/nvim/` and now the cpu and ram usage are much more reasonable.  I hope this helps because this was giving me a lot of headaches over the last couple of days too!

**Comment by jonleopard** [comment]
Thank you so much for the write-up @sazro89 ! While it didn't solve my problem, you've definitely put me on the right trail. 

Cheers :)
